### PR TITLE
Add zstd compressed payload support to RPM (from @n3npq in #256)

### LINKED ISSUE #256: Adding ZSTD (or other FSE compressor)
There are (at least) 3 new FSE compression formats that have been released in
the past few years: zstd (Facebook),  brotli (Google), lzfse (Apple).

The new compressors have better compression ratios and speed than gzip/zlib.

And while the compression ratio is still not as good as XZ/LZMA, the decompression speed (which is mostly rate determining for RPM installation), of zstd is considerably faster than XZ/LZMA.

One or more of these implementations might be useful for RPM.

(aside)
There's an implementation for RPM+ZSTD available here: [http://rpm5.org/files/rpm/rpm-5.4/SNAPSHOT/rpm-5.4.18-0.20170718.src.rpm](url) that can be adapted to RPM4. I can/will do the patch if there is sufficient interest.

### PR DESCRIPTION
**Pull request by Conan-Kudo**
This PR adds support for zstd compressed payloads for RPMs. This addresses the issues with duplicate m4 files with #283 and adds proper attribution.

Obsoletes #283 
Closes #256

### CODE REVIEW

File: build/pack.c [hunk 1]
```diff
@@ -348,6 +348,12 @@ static char *getIOFlags(Package pkg)
 	    compr = "lzma";
 	    (void) rpmlibNeedsFeature(pkg, "PayloadIsLzma", "4.4.6-1");
 #endif
+#if HAVE_ZSTD_H
+	} else if (rstreq(s+1, "zstdio")) {
+	    compr = "zstd";
+	    /* Add prereq on rpm version that understands zstd payloads */
+	    (void) rpmlibNeedsFeature(pkg, "PayloadIsZstd", "5.4.18-1");
```
Comment (ignatenkobrain) [inline build/pack.c]: I think this should be something different rather..

File: configure.ac [hunk 2]
```diff
@@ -203,6 +204,20 @@ AC_CHECK_HEADERS([lzma.h],[
 AC_SUBST(WITH_LZMA_LIB)
 
 #=================
+# Check for zstd library.
+
+dnl # Facebook Zstd
+RPM_CHECK_LIB(
```
Comment (ignatenkobrain) [inline configure.ac]: instead of this hug macro I would probably prefer to stick with pkg-config..

File: lib/rpmds.c [hunk 3]
```diff
@@ -1278,6 +1278,11 @@ static const struct rpmlibProvides_s rpmlibProvides[] = {
     { "rpmlib(RichDependencies)",    "4.12.0-1",
 	(		RPMSENSE_EQUAL),
     N_("support for rich dependencies.") },
+#if HAVE_ZSTD_H
+    { "rpmlib(PayloadIsZstd)",		"5.4.18-1",
```
Comment (ignatenkobrain) [inline lib/rpmds.c]: same here about version

File: build/pack.c (hunk 1 above)
Comment (n3npq) [inline build/pack.c]: What "zstdio" should be different? You'll have to make amore positive suggestion and state the reasoning for me to pay attention.

Comment (ignatenkobrain) [inline build/pack.c]: the version I meant ;)

Comment (n3npq) [inline build/pack.c]: If you mean the tracking dependency, well, zstd got implemented elsewhere before this patch was submitted. Change and you, not me, will introduce confusion.

File: configure.ac (hunk 2 above)
Comment (n3npq) [inline configure.ac]: RPM_CHECK_LIB uses pkgconfig (when available). I'm not sure what you wish.

File: lib/rpmds.c (hunk 3 above)
Comment (n3npq) [inline lib/rpmds.c]: The version was dictated by where first implemented. See the equivalent in xzdio/lzdio tracking dependencies.

File: build/pack.c (hunk 1 above)
Comment (Conan-Kudo) [inline build/pack.c]: This is not unusual. We did the same thing for the Xz payload. It's not worth the extra breakage to do that.

File: configure.ac (hunk 2 above)
Comment (Conan-Kudo) [inline configure.ac]: `RPM_CHECK_LIB` makes it work when pkgconfig isn't available (macOS, etc.). It also makes it easy to support internal or external libraries for rpm, which can be handy for stuff where relying on shared libraries might get dangerous.

**Comment by ignatenkobrain** [comment]
Superseded by #297

**Pull request closed without merging**